src/workers/build/mod.rs

```diff
@@ -0,0 +1,151 @@
+mod krate;
+
+use std::fs;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use super::{Resource, Script, WasmModule, Worker};
+
+use crate::commands::build::wranglerjs;
+use crate::commands::publish::package::Package;
+use crate::settings::project::{Project, ProjectType};
+use crate::terminal::message;
+use crate::{commands, install};
+
+pub fn build(project: &Project) -> Result<Worker, failure::Error> {
+    match project.project_type {
+        ProjectType::JavaScript => {
+            message::info("JavaScript project found. Skipping unnecessary build!")
+        }
+        ProjectType::Rust => {
+            let tool_name = "wasm-pack";
+            let binary_path = install::install(tool_name, "rustwasm")?.binary(tool_name)?;
+            let args = ["build", "--target", "no-modules"];
+
+            let command = command(&args, binary_path);
+            let command_name = format!("{:?}", command);
+
+            commands::run(command, &command_name)?;
+        }
+        ProjectType::Webpack => {
+            wranglerjs::run_build(project)?;
+        }
+    }
+
+    let worker = match project.project_type {
+        ProjectType::Rust => wasm_worker(&project.name)?,
+        ProjectType::Webpack => webpack_worker(&project.name)?,
+        ProjectType::JavaScript => js_worker(&project.name)?,
+    };
+    // add other resoureces
```

Comment:
this comment is confusing to me

Reply:
This is where we'll add KV namespaces once we have that configuration in the `Project` struct. I can clarify this comment or remove it.

Reply:
i'm up for the clarification or removing it- up to you!